app/connectors/GetCalculationListConnectorLegacy.scala

```diff
@@ -27,8 +27,22 @@ class GetCalculationListConnectorLegacy @Inject()(http: HttpClient, val appConfi
 
   def calcList(nino: String, taxYear: Option[String])(implicit hc: HeaderCarrier): Future[GetCalculationListResponseLegacy] = {
 
+    val localEnv = "http://localhost:9081"  // ideally in the actual implementation, we shouldn't use entire url and instead use "localhost"
+
+    val stagingEnv = "https://www.staging.tax.service.gov.uk" // ideally in the actual implementation, we shouldn't use entire url and instead use "staging"
+
+    val env = hc.headers(Seq("env")).headOption.map(_._2)
+
+    val isStagingOrLocal = env.contains(localEnv) || env.contains(stagingEnv)
```

Comment:
I think that there must be other way to detect what env this app is running on; 
in the proposed approach we reading this externally, so this can be manipulated, i.e if there are wrong values in the config ?